utils.pwsh/Utils-Patch.ps1

```diff
@@ -0,0 +1,145 @@
+    )
+
+    $Params = @(
+        '-R', '-p1', '-N'
+        $(if ( $DryRun ) { "--dry-run" })
+        $(if ( $VerbosePreference -eq 'Continue' ) { "--verbose" })
+        '-i', $PatchFile
+        $(if ( $Silent ) { " > NUL" })
+    )
+
+    Log-Information "Reverting patch $([System.IO.Path]::GetFileName($PatchFile))"
+    Run-PatchExe @Params
+}
+
+function Apply-Patch {
+    <#
+        .SYNOPSIS
+            Applies a patch
+        .DESCRIPTION
+            Allows applying the changes contained in a patch file
+        .EXAMPLE
+            Apply-Patch -PatchFile <Your-Patch-File>
+            Apply-Patch -PatchFile <Your-Patch-File> -DryRun
+    #>
+
+    param(
+        [Parameter(Mandatory)]
+        [string] $PatchFile,
+        [switch] $DryRun,
+        [switch] $Silent
+    )
+
+    $Params = @(
+        '-g', '0', '-f', '-p1'
+        $(if ( $DryRun ) { "--dry-run" })
+        $(if ( $VerbosePreference -eq 'Continue' ) { "--verbose" })
+        '-i', $PatchFile
+        $(if ( $Silent ) { " > NUL" })
+    )
+
+    Log-Information "Applying patch $([System.IO.Path]::GetFileName($PatchFile))"
+    Run-PatchExe @Params
+}
+
+function Safe-Patch {
+    <#
+        .SYNOPSIS
+            Applies a patch with hash checking
+        .DESCRIPTION
+            Allows applying a patch checked against a provided SHA256 checksum.
+        .EXAMPLE
+            Safe-Patch -PatchFile <You-Patch-File> -HashSum <HashSum>
+            Safe-Patch -PatchFile <You-Patch-File> -HashSum <HashSum> -Path <Path-To-Apply-Patch-To>
+    #>
+
+    param(
+        [Parameter(Mandatory)]
+        [string] $PatchFile,
+        [Parameter(Mandatory)]
+        [string] $HashSum,
+        [string] $Path
+    )
+
+    if ( ! ( Test-Path function:Log-Information ) ) {
+        . $PSScriptRoot/Util-Logger.ps1
+    }
+
+    if ( $PatchFile.SubString(0, 5) -eq "https" ) {
+        $WebRequestParams = @{
+            UserAgent = "NativeHost"
+            Uri = $PatchFile
+            OutFile = [System.IO.Path]::GetFileName($PatchFile)
+            UseBasicParsing = $true
+            ErrorAction = "Stop"
+        }
+
+        Invoke-WebRequest @WebRequestParams
+    } elseif  ( ! ( Test-Path $PatchFile ) ) {
+        throw "Supplied patch file ${PatchFile} not found"
+    }
+
+    Push-Location -Stack SafePatchTemp
+
+    $PatchHash = (Get-FileHash -Path $PatchFile -Algorithm SHA256).Hash
+
+    if ( $HashSum -eq $PatchHash ) {
+        Log-Information "Hash of patch file $([System.IO.Path]::GetFileName($PatchFile)) confirmed as '${HashSum}'"
+
+        if ( $Path -ne "" ) {
+            Set-Location -Path $Path
+        }
+
+        try {
+            Revert-Patch -DryRun -Silent -PatchFile $PatchFile
+        } catch {
+            Apply-Patch -PatchFile $PatchFile
+        }
+
+        Pop-Location -Stack SafePatchTemp
+    } else {
+        Pop-Location -Stack SafePatchTemp
+
+        throw "Hash of patch file ${PatchFile} is '${PatchHash}' - expected '${HashSum}'"
```

Comment:
I'd kind of prefer two sentences (". Expected "), or ", but expected" rather than a hyphen.